src/msgraph_core/requests/batch_request_builder.py

```diff
@@ -0,0 +1,142 @@
+from typing import TypeVar, Type, Dict, Optional, Union
+import json
+import logging
+
+from kiota_abstractions.request_adapter import RequestAdapter
+from kiota_abstractions.request_information import RequestInformation
+from kiota_abstractions.method import Method
+from kiota_abstractions.serialization import Parsable
+from kiota_abstractions.headers_collection import HeadersCollection
+from kiota_abstractions.api_error import APIError
+
+from .batch_request_content import BatchRequestContent
+from .batch_request_content_collection import BatchRequestContentCollection
+from .batch_response_content import BatchResponseContent
+from .batch_response_content_collection import BatchResponseContentCollection
+
+T = TypeVar('T', bound='Parsable')
+
+APPLICATION_JSON = "application/json"
+
+
+class BatchRequestBuilder:
+    """
+    Provides operations to call the batch method.
+    """
+
+    def __init__(
+        self,
+        request_adapter: RequestAdapter,
+        error_map: Optional[Dict[str, Type[Parsable]]] = None
+    ):
+        if request_adapter is None:
+            raise ValueError("request_adapter cannot be Null.")
+        self._request_adapter = request_adapter
+        self.url_template = f"{self._request_adapter.base_url}/$batch"
+        self.error_map = error_map or {}
+
+    async def post(
+        self,
+        batch_request_content: Union[BatchRequestContent, BatchRequestContentCollection],
+        response_type: Optional[Type[T]] = None,
```

Comment:
Isn't this where we set the details of the request, and the deserialization class is expected when calling send_async?

Reply:
if we don't pass it here, how will we get it, to be used in the API call

Reply:
> Isn't this where we set the details of the request, and the deserialization class is expected when calling send_async?

The larger response from the batch endpoint will always be of the same schema (i.e. BatchResponseContent) so that is what would be passed to the send method without the user having to specify it. 

The larger batch message will contain several different individual responses, so if only one type is a passed, it means that the individual response may be unable to be deserialized into different response types. 

A type parameter should ideally be passed on the response content. See usage examples for dotnet at https://github.com/microsoftgraph/msgraph-sdk-design/blob/main/content/BatchResponseContent.md#usage-examples

```cs
EventCollectionResponse events = await batchResponseContent.GetResponseByIdAsync<EventCollectionResponse>(eventsRequestId);
```

Reply:
This is implemented:

```python
 individual_response = batch_response_content.get_response_by_id(
            batch_request_item1.id, User
        )
```